src/container/tenant_builder.rs

```diff
@@ -253,70 +280,84 @@ impl<'a> TenantContainerBuilder<'a> {
             let caps: SpecCapabilities =
                 caps.iter().map(|c| SpecCapability::from_cap(*c)).collect();
 
-            if let Some(ref mut spec_caps) = spec
-                .process
-                .as_mut()
+            if let Some(ref spec_caps) = spec
+                .process()
+                .as_ref()
                 .context("no process in spec")?
-                .capabilities
+                .capabilities()
             {
-                spec_caps
-                    .ambient
-                    .as_mut()
-                    .context("no ambient caps in process spec")?
-                    .extend(&caps);
-                spec_caps
-                    .bounding
-                    .as_mut()
-                    .context("no bounding caps in process spec")?
-                    .extend(&caps);
-                spec_caps
-                    .effective
-                    .as_mut()
-                    .context("no effective caps in process spec")?
-                    .extend(&caps);
-                spec_caps
-                    .inheritable
-                    .as_mut()
-                    .context("no inheritable caps in process spec")?
-                    .extend(&caps);
-                spec_caps
-                    .permitted
-                    .as_mut()
-                    .context("no permitted caps in process spec")?
-                    .extend(&caps);
-            } else {
-                spec.process
-                    .as_mut()
-                    .context("no process in spec")?
-                    .capabilities = Some(LinuxCapabilities {
-                    ambient: caps.clone().into(),
-                    bounding: caps.clone().into(),
-                    effective: caps.clone().into(),
-                    inheritable: caps.clone().into(),
-                    permitted: caps.into(),
-                })
+                let mut cb = LinuxCapabilitiesBuilder::default();
```

Comment:
can we change to `capability_builder`? `cb` gets confused to `call back`. Or just builder.